geomstats/information_geometry/gamma.py

```diff
@@ -0,0 +1,1200 @@
+            state = gs.stack([position, velocity])
+            vel, acc = self.geodesic_equation(state, _)
+            eq = (vel, acc)
+            return gs.hstack(eq)
+
+        def path(t):
+            """Generate parameterized function for geodesic curve.
+
+            Parameters
+            ----------
+            t : array-like, shape=[n_times,]
+                Times at which to compute points of the geodesics.
+
+            Returns
+            -------
+            geodesic : array-like, shape=[..., n_times, dim]
+                Values of the geodesic at times t.
+            """
+            t = gs.to_ndarray(t, to_ndim=1)
+            n_times = len(t)
+            geod = []
+
+            if n_times < n_steps:
+                t_int = gs.linspace(0, 1, n_steps + 1)
+                tangent_vecs = gs.einsum("i,...k->...ik", t, initial_tangent_vec)
+                for point, vec in zip(initial_point, tangent_vecs):
+                    point = gs.tile(point, (n_times, 1))
+                    exp = []
+                    for pt, vc in zip(point, vec):
+                        initial_state = gs.hstack([pt, vc])
+                        solution = odeint(ivp, initial_state, t_int, ())
+                        exp.append(solution[-1, : self.dim])
+                    exp = exp[0] if n_times == 1 else gs.stack(exp)
+                    geod.append(exp)
+            else:
+                t_int = t
+                for point, vec in zip(initial_point, initial_tangent_vec):
+                    initial_state = gs.hstack([point, vec])
+                    solution = odeint(ivp, initial_state, t_int, ())
+                    geod.append(solution[:, : self.dim])
+
+            geod = geod[0] if len(initial_point) == 1 else gs.stack(geod)
+            return gs.where(geod < gs.atol, gs.atol, geod)
+
+        return path
+
+    def _exp_ivp(self, tangent_vec, base_point, n_steps=N_STEPS):
+        """Compute the exponential map.
+
+        Comute the exponential map associated to the Fisher information metric
+        by solving the initial value problem associated to the geodesic
+        ordinary differential equation (ODE) using the Christoffel symbols.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., dim]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., dim]
+            Base point.
+        n_steps : int
+            Number of steps for integration.
+            Optional, default: 100.
+
+        Returns
+        -------
+        exp : array-like, shape=[..., dim]
+            End point of the geodesic starting at base_point with
+            initial velocity tangent_vec and stopping at time 1.
+        """
+        stop_time = 1.0
+        geodesic = self._geodesic_ivp(base_point, tangent_vec, n_steps)
+        exp = geodesic(stop_time)
+
+        return exp
+
+    def _approx_geodesic_bvp(
```

Comment:
This function should be tested, see e.g. [here](https://github.com/geomstats/geomstats/blob/9bad7b8cc618fc0b9cc3dee13d01698a8f86bb11/tests/tests_geomstats/test_dirichlet.py#L216).